amis/packer_alinux.json

```diff
@@ -13,13 +13,15 @@
     "instance_type" : "{{env `AWS_FLAVOR_ID`}}",
     "subnet_id" : "{{env `AWS_SUBNET_ID`}}",
     "vpc_id" : "{{env `AWS_VPC_ID`}}",
-    "ami_name_prefix" : "{{env `AMI_NAME_PREFIX`}}"
+    "ami_name_prefix" : "{{env `AMI_NAME_PREFIX`}}",
+    "region" : "{{env `AWS_REGION`}}",
+    "source_ami" : "{{env `ALINUX_IMAGE_ID`}}"
   },
   "builders" : [
     {
       "type" : "amazon-ebs",
-      "region" : "us-east-1",
-      "source_ami" : "ami-55ef662f",
+      "region" : "{{user `region`}}",
+      "source_ami" : "{{user `source_ami`}}",
```

Comment:
Just asking, was not possible to leave a default like
"source_ami" : "{{user `source_ami`}}" | "ami-55ef662f",

Reply:
@lukeseawalker aforementioned PR was never merged. Looks like it's still not supported

Reply:
ok, pity. I think it's better to move the defaults into the calling script, build_ami.sh, adding a new input parameter "build_region" and renaming the old "region" parameter to "target_region"

```
os=$1
region=$2
public=$3
build_date=$4
```
become
```
os=$1
build_region=$2
target_region=$3
public=$4
build_date=$5
```

In this way we can have build_region = us-east-1 or PDT as input parameter. Then we could check if ALINUX_IMAGE_ID or the others are not set and set the default value based on build_region. Something like
```
if ALINUX_IMAGE_ID not set {
   if build_region == us-east-1 then ALINUX_IMAGE_ID = ami-55ef662f
   if build_region  == PDT then ALINUX_IMAGE_ID = ami-xxx
}

if ALINUX_IMAGE_ID empty, then error
Repeat for the others *_IMAGE_ID
```

In this way we can have the default value for the image id (based on the build_region) but also the possibility to override that value, if needed.
What do you think @sean-smith?

Reply:
I'd prefer to have a seperate file, like `default/attributes.rb` that just contains the ami id's. That'll allow us to programmatically update the base ami easily.

Reply:
ok for the separate file, but we are not in the "domain" where we can use .rb files, as we are in the "packer" domain. There is a "packer_variables.json" file where we can store variable values, but as long as the packer syntax could not give as the possibility to use a value and if this value is not set to fallback to a default value, I'd prefer to move this logic to the calling script. Then we can use a properties file which contain the default values for the amis and source this file from the script. Something like
```
packer_amis.properties

US_EAST_1_DEFAULT_ALINUX_IMAGE_ID=ami-55ef662f
GOVCLOUD_DEFAULT_ALINUX_IMAGE_ID=ami-xxx
```
then from the build_ami.sh
```
source packer_amis.properties

if ALINUX_IMAGE_ID not set {
   if build_region == us-east-1 then ALINUX_IMAGE_ID = ${US_EAST_1_DEFAULT_ALINUX_IMAGE_ID}
   if build_region  == PDT then ALINUX_IMAGE_ID = ${GOVCLOUD_DEFAULT_ALINUX_IMAGE_ID}
}

if ALINUX_IMAGE_ID empty, then error
Repeat for the others *_IMAGE_ID
```

Reply:
I'd much prefer to use `source_ami_filter` and not explicitly use the ami name. https://www.packer.io/docs/builders/amazon-ebs.html#source_ami_filter